Packs/Gatewatcher-AionIQ/Integrations/GCenter103/GCenter103.py

```diff
@@ -0,0 +1,851 @@
+        query = query_empty_selected_engines_builder(from_to=from_to, max_fetch=max_fetch)
+        gw_metadata = handle_little_fetch_metadata(client=client, query=query, fetch_type=fetch_type)
+        incidents_m = []
+        if len(gw_metadata) > 0:
+            incidents_m = index_metadata_incidents(to_index=gw_metadata, incidents=incidents)
+
+        return incidents_a + incidents_m
+
+def fetch_empty_selected_engines(client: GwClient, 
+                                 max_fetch: int, 
+                                 fetch_type: str, 
+                                 incidents, 
+                                 params: dict[str, Any]):
+
+    from_to: list[str] = last_run_range(params=params)
+    query: dict[str, Any] = query_empty_selected_engines_builder(from_to=from_to, max_fetch=max_fetch)
+
+    if max_fetch > 10000:
+
+        gw_alerts = handle_big_fetch_empty_selected_engines(
+            client=client, query=query, max_fetch=max_fetch, fetch_type=fetch_type)
+        incidents_a = index_alerts_incidents(to_index=gw_alerts, incidents=incidents, params=params)
+
+        query = query_empty_selected_engines_builder(from_to=from_to, max_fetch=max_fetch)
+        gw_metadata = handle_big_fetch_metadata(client=client, query=query, max_fetch=max_fetch, fetch_type=fetch_type)
+        incidents_m = index_metadata_incidents(to_index=gw_metadata, incidents=incidents)
+
+        return incidents_a + incidents_m
+
+    else:
+
+        gw_alerts = handle_little_fetch_empty_selected_engines(client=client, query=query, fetch_type=fetch_type)
+        incidents_a = []
+        if len(gw_alerts) > 0:
+            incidents_a = index_alerts_incidents(to_index=gw_alerts, incidents=incidents, params=params)
+
+        query = query_empty_selected_engines_builder(from_to=from_to, max_fetch=max_fetch)
+        gw_metadata = handle_little_fetch_metadata(client=client, query=query, fetch_type=fetch_type)
+        incidents_m = []
+        if len(gw_metadata) > 0:
+            incidents_m = index_metadata_incidents(to_index=gw_metadata, incidents=incidents)
+
+        return incidents_a + incidents_m
+
+def fix_broken_list(params: dict[str, Any]) -> list[str]:
+
+    e_s: list[str] = []
+    broken_demisto_list = params['engine_selection']
+    bdl = broken_demisto_list
+
+    if "malcore" in str(bdl):
+        e_s.append("malcore")
+    if "shellcode_detect" in str(bdl):
+        e_s.append("shellcode_detect")
+    if "malicious_powershell_detect" in str(bdl):
+        e_s.append("malicious_powershell_detect")
+    if "sigflow_alert" in str(bdl):
+        e_s.append("sigflow_alert")
+    if "dga_detect" in str(bdl):
+        e_s.append("dga_detect")
+    if "active_cti" in str(bdl):
+        e_s.append("ioc")
+    if "retrohunt" in str(bdl):
+        e_s.append("retrohunt")
+    if "ransomware_detect" in str(bdl):
+        e_s.append("ransomware_detect")
+    if "beacon_detect" in str(bdl):
+        e_s.append("beacon_detect")
+
+    return e_s
```

Comment:
You can try something like this:

```suggestion
def fix_broken_list(params: Dict[str, Any]) -> List[str]:
    if 'engine_selection' not in params or not isinstance(params['engine_selection'], (str, list)):
        raise ValueError("Invalid 'engine_selection' parameter")

    bdl = params['engine_selection']

    known_engines = {
        "malcore", "shellcode_detect", "malicious_powershell_detect",
        "sigflow_alert", "dga_detect", "active_cti", "retrohunt",
        "ransomware_detect", "beacon_detect"
    }
    e_s = []

    if isinstance(bdl, str):
        e_s = [engine for engine in known_engines if engine in bdl]
    elif isinstance(bdl, list):
        e_s = [engine for engine in known_engines if engine in bdl]

    return e_s
```